ambry-tools/src/main/java/com.github.ambry/store/DumpData.java

```diff
@@ -562,40 +559,31 @@ public void compareReplicaIndexEntriestoLogContent(String replicaRootDirectory,
     RandomAccessFile randomAccessFile = new RandomAccessFile(new File(logFile), "r");
     logger.info("Comparing Index entries to Log ");
     File replicaDirectory = new File(replicaRootDirectory);
-    MergedIntervals coveredRangesInLog = new MergedIntervals();
-    Map<Long, Long> deletedMsgInfo = new HashMap<>();
     for (File indexFile : replicaDirectory.listFiles(new IndexFileNameFilter())) {
-      compareIndexEntriesToLogContent(indexFile, replicaDirectory, randomAccessFile, coveredRangesInLog, deletedMsgInfo,
-          false);
+      TreeMap<Long, Long> coveredRanges = new TreeMap<Long, Long>();
+      Set<Long> deleteMsgsOriginalOffsets = new HashSet<Long>();
+      compareIndexEntriesToLogContent(indexFile, replicaDirectory, randomAccessFile, coveredRanges,
+          deleteMsgsOriginalOffsets);
     }
```

Comment:
The "holes at the end" problem is ok (we can find a way around it) - but I think checking that all the index segments together cover the whole log is a requirement.